components/mapbox/actions/generate-directions/generate-directions.mjs

```diff
@@ -0,0 +1,59 @@
+import mapbox from "../../mapbox.app.mjs";
+
+export default {
+  key: "mapbox-generate-directions",
+  name: "Generate Directions",
+  description: "Generates directions between two or more locations using Mapbox API. [See the documentation](https://docs.mapbox.com/api/navigation/directions/).",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    mapbox,
+    startCoordinate: {
+      type: "string",
+      label: "Start Coordinate",
+      description: "The starting point in the format `longitude,latitude`",
+    },
+    endCoordinate: {
+      type: "string",
+      label: "End Coordinate",
+      description: "The ending point in the format `longitude,latitude`",
+    },
+    transportationMode: {
+      propDefinition: [
+        mapbox,
+        "transportationMode",
+      ],
+    },
+    steps: {
+      type: "boolean",
+      label: "Steps",
+      description: "Whether to return steps and turn-by-turn instructions (`true`) or not (`false`, default)",
```

Comment:
If there is a default, this prop should be optional - or if it's not optional, it shouldn't mention a default